recipes/libexif/all/conanfile.py

```diff
@@ -57,56 +52,63 @@ def configure(self):
         del self.settings.compiler.cppstd
 
     def build_requirements(self):
-        self.build_requires("gettext/0.21")
-        self.build_requires("libtool/2.4.6")
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
-
-    def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+        self.tool_requires("gettext/0.21")
+        self.tool_requires("libtool/2.4.6")
+        if self._settings_build.os == "Windows":
+            if not self.conf.get("tools.microsoft.bash:path", default=False, check_type=bool):
+                self.tool_requires("msys2/cci.latest")
+            self.win_bash = True
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
-
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
-        self._autotools = AutoToolsBuildEnvironment(self, win_bash=self._settings_build.os == "Windows")
-        self._autotools.libs = []
-        if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) >= "12":
-            self._autotools.flags.append("-FS")
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = AutotoolsToolchain(self)
         yes_no = lambda v: "yes" if v else "no"
-        args = [
+        tc.configure_args.extend([
             "--enable-shared={}".format(yes_no(self.options.shared)),
             "--enable-static={}".format(yes_no(not self.options.shared)),
             "--disable-docs",
             "--disable-nls",
             "--disable-rpath",
-        ]
-        self._autotools.configure(args=args, configure_dir=self._source_subfolder)
-        return self._autotools
+        ])
+        if is_msvc(self) and Version(self.settings.compiler.version) >= "12":
```

Comment:
Need to account for `msvc` here.

```suggestion
        if (self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version) >= "12") or \
                (self.settings.compiler == "msvc" and Version(self.settings.compiler.version) >= "180"):
```